Add defense mode with toggle button

coderkevin opened the pull request:
This adds a defense mode which is activated/deactivated with a toggle button

coderkevin committed:
Add defense mode with toggle button

coderkevin committed:
Stop the defense solenoid 100 ms after use

coderkevin committed:
Re-add widgets to drive tab on show

Previously, widgets would sometimes be disabled on the dashboard. With
this change, the robot can be re-enabled and it will re-add the widgets
to help ensure they show up correctly

Stick-2357 reviewed (approved):
Looks good. We might want to move more preset code into ArmSub, so we don't have to set last preset manually, but what we have here works.

coderkevin committed:
Fix DriveTab

The previous commit took out the if statements around the DriveTab
elements, but trying to re-add widgets causes an exception
